src/modules/geometry.cxx

```diff
@@ -0,0 +1,28 @@
+module;
+
+export module Geometry;
+
+export class Vector_2d
+{
+public:
+    double x_component, y_component;
+    Vector_2d(double x, double y) : x_component(x), y_component(y) {}
+
+};
+
+export class Point
+{
+public:
+    double x, y;
+    Point(int x, int y) : x(x), y(y) {}
+};
+
+export class Ray
+{
+public:
+    Point origin;
+    Vector_2d velocity;
```

Comment:
We are stuck in 2D. We also need 3D vectors and points. How could this be done with minimal duplication?